src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -112,36 +121,160 @@ class ClipBam
       out += rec
     }
 
+    this.metrics.foreach { path =>
```

Comment:
I think this might be clearer as:
```scala
(this.metrics, this.metricsMap) match {
  case (Some(path), Some(ms)) =>
    ....
  case _ => Unit
}
```

I had to stare for a minute to remember that metricMap is an `Option[Map[]]` and that's why you needed the foreach.  Also, having the second foreach use `metrics` as it's parameter name is confusing since you have an instance variable called `metrics`.

Reply:
Done